types/lib/model.d.ts

```diff
@@ -106,16 +106,21 @@ export interface ScopeOptions {
   method: string | readonly [string, ...unknown[]];
 }
 
+type AllowOrAnd<T> = T | { [Op.or]: T[] } | { [Op.and]: T[] };
+type AllowOrAndRecursive<T> = T | { [Op.or]: AllowArray<AllowOrAndRecursive<T>> } | { [Op.and]: AllowArray<AllowOrAndRecursive<T>> };
```

Comment:
I think this part may be incorrect. I need to double check if the following query is valid:

```typescript
User.findAll({
  where: {
    // can we have and & or in the same object? Probably a yes
    [Op.and]: [{ id: 1, id: 2 }],
    [Op.or]: [{ id: 3, id: 4 }],
  }
})
```

Reply:
I've added a test and this works both in typings and when generating the query